plugin-annotation/src/main/java/com/mapbox/mapboxsdk/plugins/annotation/AnnotationManager.java

```diff
@@ -222,7 +226,18 @@ public void update(List<T> annotationList) {
    */
   public void updateSource() {
     draggableAnnotationController.onSourceUpdated();
-    internalUpdateSource();
+    postUpdateSource();
+  }
+
+  void postUpdateSource() {
+    // Only schedule a new refresh if not already scheduled
+    if (isSourceUpToDate) {
+      isSourceUpToDate = false;
+      mapView.post(() -> {
+        internalUpdateSource();
+        isSourceUpToDate = true;
```

Comment:
As `isSourceUpToDate` is accessed from multiple threads, I suggest you make sure this is thread safe and use either `volatile` + `synchronized` or `AtomicBoolean` concepts.

Reply:
:+1:, see https://github.com/maplibre/maplibre-plugins-android/pull/12/commits/2fbbb4607fef6f086041413c883d2d260e9ccdb3